soccer/Robot.cpp

```diff
@@ -22,55 +23,57 @@ using Planning::MotionInstant;
 
 /** thresholds for avoidance of opponents - either a normal (large) or an
  * approach (small)*/
-const float Opp_Avoid_Small = Robot_Radius - 0.03;
+constexpr float Opp_Avoid_Small_Thresh = 0.03;
+constexpr float Opp_Avoid_Small = Robot_Radius - Opp_Avoid_Small_Thresh;
 /** threshold for avoiding the ball*/
-const float Ball_Avoid_Small = 2.0 * Ball_Radius;
+constexpr float Ball_Avoid_Small_Mult = 2.0;
+constexpr float Ball_Avoid_Small = Ball_Avoid_Small_Mult * Ball_Radius;
 /**
  * When verbose is true, a lot of extra debug info is printed
  * to the console about path planning, etc
  */
-const bool verbose = false;
+constexpr bool verbose = false;
 
-Robot::Robot(Context* context, unsigned int shell, bool self)
+Robot::Robot(Context* context, int shell, bool self)
     : _context(context), _shell(shell), _self(self) {}
 
 #pragma mark OurRobot
 
 REGISTER_CONFIGURABLE(OurRobot)
 
-ConfigDouble* OurRobot::_selfAvoidRadius;
-ConfigDouble* OurRobot::_oppAvoidRadius;
-ConfigDouble* OurRobot::_oppGoalieAvoidRadius;
-ConfigDouble* OurRobot::_dribbleOutOfBoundsOffset;
+ConfigDouble* OurRobot::self_avoid_radius;
+ConfigDouble* OurRobot::opp_avoid_radius;
+ConfigDouble* OurRobot::opp_goalie_avoid_radius;
+ConfigDouble* OurRobot::dribble_out_of_bounds_offset;
 
 void OurRobot::createConfiguration(Configuration* cfg) {
-    _selfAvoidRadius =
-        new ConfigDouble(cfg, "PathPlanner/selfAvoidRadius", Robot_Radius);
-    _oppAvoidRadius = new ConfigDouble(cfg, "PathPlanner/oppAvoidRadius",
-                                       Robot_Radius - 0.01);
-    _oppGoalieAvoidRadius = new ConfigDouble(
-        cfg, "PathPlanner/oppGoalieAvoidRadius", Robot_Radius + 0.05);
-    _dribbleOutOfBoundsOffset = new ConfigDouble(
-        cfg, "PathPlanner/dribbleOutOfBoundsOffset", 0.05);
-}
-
-OurRobot::OurRobot(Context* context, int shell) : Robot(context, shell, true) {
-    _cmdText = new std::stringstream();
+    constexpr float kOppAvoidRadiusThresh = 0.01;
+    constexpr float kOppGoalieAvoidRadiusBuffer = 0.05;
+    constexpr float kDribbleOutOfBoundsOffset = 0.01;
 
+    self_avoid_radius =
+        new ConfigDouble(cfg, "PathPlanner/selfAvoidRadius", Robot_Radius);
+    opp_avoid_radius = new ConfigDouble(cfg, "PathPlanner/oppAvoidRadius",
+                                        Robot_Radius - kOppAvoidRadiusThresh);
+    opp_goalie_avoid_radius =
+        new ConfigDouble(cfg, "PathPlanner/oppGoalieAvoidRadius",
+                         Robot_Radius + kOppGoalieAvoidRadiusBuffer);
+    dribble_out_of_bounds_offset = new ConfigDouble(
+        cfg, "PathPlanner/dribbleOutOfBoundsOffset", kDribbleOutOfBoundsOffset);
```

Comment:
I'm not sure I fully agree with this change. To me it doesn't add any extra information to these default config values

Reply:
oh I think I can revert this change. Originally I tried to use `unique_ptr`, but that caused some bugs so I was debugging